### PR TITLE
BoothRegistration

### PR DESCRIPTION
**Pull request by romo-odoo**

### CODE REVIEW

File: addons/website_event_booth/static/src/interactions/booth_register.js [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+        ".o_wbooth_registration_submit": {
+            "t-on-click.prevent": this.onClickSubmit,
+        },
+        ".o_wbooth_registration_confirm": {
+            "t-on-click.prevent.stop": (ev) => this.onClickConfirm(ev.currentTarget),
+        },
+    };
+
+    start() {
+
```
Comment (bso-odoo) [inline addons/website_event_booth/static/src/interactions/booth_register.js]: blank line

File: addons/website_event_booth/static/src/interactions/booth_register.js [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+        this.boothCache = {};
+        this.isFirstRender = true;
+
+        this.eventId = parseInt(this.el.dataset.eventId);
+
+        this.activeBoothCategoryId = false;
+        this.selectedBoothIds = [];
+        this.selectedBoothCategory = this.el.querySelector('input[name="booth_category_id"]:checked');
```
Comment (bso-odoo) [inline addons/website_event_booth/static/src/interactions/booth_register.js]: Define variables in `setup`

File: addons/website_event_booth/static/src/interactions/booth_register.js [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+        targetEl.closest(".form-check").classList.remove("text-danger");
+        this.updateUIBoothCategoryChanged(this.countSelectedBooths());
+    }
+
+    async onClickSubmit() {
+        const selectedBoothEls = this.el.querySelectorAll("input[name=event_booth_ids]:checked");
+        const selectedBoothIds = [];
+        for (const selectedBoothEl of selectedBoothEls) {
+            selectedBoothIds.push(parseInt(selectedBoothEl.value));
+        }
```
Comment (bso-odoo) [inline addons/website_event_booth/static/src/interactions/booth_register.js]: This seems more expensive than the old `map` operation.

File: addons/website_event_booth/static/src/interactions/booth_register.js [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+        if (errors.includes('boothError')) {
+            errorMessages.push(_t("Booth registration failed."));
+        }
+        if (errors.includes('boothCategoryError')) {
+            errorMessages.push(_t("The booth category doesn't exist."));
+        }
+
+        const errorMessageEl = this.el.querySelector(".o_wbooth_registration_error_message");
+        errorMessageEl.textContent = errorMessages.join(" ");
```
Comment (bso-odoo) [inline addons/website_event_booth/static/src/interactions/booth_register.js]: Can't this be a `t-out` ?

File: addons/website_event_booth/static/src/interactions/booth_register.js [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+
+    /**
+     * Display the errors with a custom message when confirming
+     * the registration if there is any.
+     *
+     * @param errors
+     */
+    updateErrorDisplay(errors) {
+        const errorSectionEl = this.el.querySelector(".o_wbooth_registration_error_section");
+        errorSectionEl.classList.toggle("d-none", !errors.length);
```
Comment (bso-odoo) [inline addons/website_event_booth/static/src/interactions/booth_register.js]: Can't this be a `t-att-class` ?

**Review by bso-odoo — COMMENTED** [review]
I think some stuff could be moved to `dynamicContent`.
Also, `updateUI` should not be in method names: they are supposed to update data structures, and `updateContent` will apply them in the DOM.

File: addons/website_event_booth/static/src/interactions/booth_register.js (hunk 4 above)
Comment (romo-odoo) [reply addons/website_event_booth/static/src/interactions/booth_register.js]: I'm not sure since we trigger an event for the change after this.

**Pull request merged**